### ISSUE TITLE
cpunum_get_localtime() called for invalid cpu num!

### ISSUE THREAD

**Issue by mahoneyt944** [libretro/mame2003-plus-libretro#1172]
Looks like we are getting a false positive since almost all games trigger this error at init with debug logging enabled. I'm guessing we are probably sending or comparing a value off by 1. These verify checks are here though: https://github.com/libretro/mame2003-plus-libretro/blob/bc288b7b5f56c8a2ca4c84001d21120147fefa96/src/cpuexec.c#L79-#L91

Would be nice to find the source of this eye sore.  `cpunum_get_localtime`

```
[MAME 2003+] Succesfully loaded ROMs.
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
cpunum_get_localtime() called for invalid cpu num!
[MAME 2003+] Perfect interleave = 0.000000160, smallest = 0.000000100
```

**Comment by grant2258** [comment]
yea but your not checking the cpunum >= cpu_gettotalcpu()

**Comment by mahoneyt944** [comment]
Once loaded it reads
```
ret: cpunum:0   total:2
ret: cpunum:1   total:2
ret: cpunum:1   total:2
void: cpunum:1   total:2
ret: cpunum:0   total:2
ret: cpunum:0   total:2
void: cpunum:1   total:2
ret: cpunum:0   total:2
ret: cpunum:0   total:2
ret: cpunum:1   total:2
ret: cpunum:1   total:2
ret: cpunum:1   total:2
ret: cpunum:1   total:2
void: cpunum:1   total:2
ret: cpunum:0   total:2
ret: cpunum:0   total:2
```

**Comment by mahoneyt944** [comment]
Looks good.

**Comment by mahoneyt944** [comment]
I ended up doing this because it's a valid error to check against for other calls. For the local time, it's not really an error to return 0 when there's no active cpu.
https://github.com/libretro/mame2003-plus-libretro/commit/763aa216d7d97b75199eef5a0f907a59e8d76edf

I noticed the timing issue as soon as I played umk3p, some of the animations would just be skipped randomly. Most noticably, when you'd uppercut your opponent it would skip the player flinging through the air and they would pop flat on the ground. Oddly it wasn't consistent either, some uppercuts would animate normally then it would skip on another. Some sounds, fight rounds  , etc would skip timings.
